docs/configuration.md

```diff
@@ -2,72 +2,68 @@
-  - [Public Network](#public-network)
-  - [Floating IPs](#floating-ips)
-  - [Operating System Images](#operating-system-images)
+- [Required configuration](#required-configuration)
+  - [Operating system image](#operating-system-image)
+  - [Public network](#public-network)
+  - [Floating IP](#floating-ip)
+  - [SSH authorized key](#ssh-authorized-key)
+  - [OpenStack credential](#openstack-credential)
+  - [Avaialability zone](#avaialability-zone)
+  - [DNS server](#dns-server)
+- [Optional Configuration](#optional-configuration)
   - [Network Filters](#network-filters)
   - [Multiple Networks](#multiple-networks)
   - [Subnet Filters](#subnet-filters)
   - [Tagging](#tagging)
   - [Metadata](#metadata)
-- [Optional Configuration](#optional-configuration)
   - [Boot From Volume](#boot-from-volume)
   - [Timeout settings](#timeout-settings)
   - [Custom pod network CIDR](#custom-pod-network-cidr)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-# Required Configuration
+# Required configuration
 
-To successfully run a Kubernetes cluster in OpenStack, you will need to configure a few essential properties. The following configurations are necessary:
-  - public network
-  - floating ip address
-  - at least one of the supported operating system images
-  - ssh authorized key
+The cluster configuration file can be generated by using [`clusterctl config cluster`](https://cluster-api.sigs.k8s.io/user/quick-start.html#generating-the-cluster-configuration) command.
+This command actually uses [the template file](../templates/cluster-template.yaml) and replace the values surrounded by `${}` with environment variables. You have to set all required environment variables in advance. The following sections explain some more details about what should be configured.
 
-## Cluster Configuration YAML files
+Note: You can use [the template file](../templates/cluster-template.yaml) by manually replacing values.
 
-The cluster configuration file can be generated by using [`clusterctl config cluster`](https://cluster-api.sigs.k8s.io/user/quick-start.html#generating-the-cluster-configuration) command or by using one of the template files placed under `/template` directory. The file contain configuration on what OpenStack elements to use to create the cluster on, and which cluster components to create. However the template is incomplete and needs to be filled in. The following sections explain some more details about what can be configured.
+## Operating system image
 
-## Public Network
+We currently depend on an update version of cloud-init otherwise the operating system choice is yours. The kubeadm bootstrap provider we're using also depends on some pre-installed software like a controller-runtime, kubelet, kubeadm, etc.. . For an examples how to build such an image take a look at [image-builder (openstack)](https://image-builder.sigs.k8s.io/capi/providers/openstack.html).
```

Comment:
```suggestion
We currently depend on an up-to-date version of cloud-init otherwise the operating system choice is yours. The kubeadm bootstrap provider we're using also depends on some pre-installed software like a container runtime, kubelet, kubeadm, etc.. . For an examples how to build such an image take a look at [image-builder (openstack)](https://image-builder.sigs.k8s.io/capi/providers/openstack.html).
```